# Pyladies meetup <img src="https://raw.githubusercontent.com/pyladies/pyladies-assets/master/geek/png/pylady_geek_partial.png" style="display:inline" width="150"  align="right">
# 1. házi feladat megoldása 

## Geeky Girl Bergengóciában

Geeky szabadságra ment. Kivételes, kissé bohókás helyre vágyott, barátai tanácsára ezért Bergengóciába vett repülőjegyet.    
A főváros, Koala Lumpia főterén egy hatalmas emlékművet pillantott meg: három koronás alak ünnepélyesen kezet fog. A talapzatra vésett feliratból kiderült, hogy a szoborcsoport Bergengócia alapító atyjait:    

__Gündzselizeben, Krakatakiwimanga és Alfadelagunix__

királyokat ábrázolja.

Geeky megcsinálta a kötelező szelfiket, majd leintett egy taxit, hogy a szállodájába menjen.

A recepción bejelentkezett a 42-es szobába, és a kulcsát kérte. A recepciós megrázta a fejét:   
"Kulcsunk nincs, de az ajtót nyitó kód nem más, mint alapító atyáink nevei hosszának szorzata plusz a szobaszám."
Geeky fellélegzett - milyen hasznos, hogy olyan gyakran szelfizik! 


### <font color="lime">1. feladat - segíts Geekynek, és számítsd ki az ajtónyitó kódot!</font>


In [1]:
ajtonyito_kod = len('Gündzselizeben') * len('Krakatakiwimanga') * len('Alfadelagunix') + 42
ajtonyito_kod

2954

### <font color="red">>>> 1. feladat megoldás - A 42-es szoba ajtónyitó kódja: 2954</font>

Geeky belépett. Az asztalon színes táblácska állt, ezzel a szöveggel:   

"HÁZIREND 

Minden vendég 13 papucsot és 11 törülközöt használhat.    
A medence 6.12-től 21.41-ig tart nyitva.    
Az étteremben minden fogás 1.37 bergengóc garasba (BEG) kerül.    
Az edzőközpontot azok használhatják, akik 5.02 lábnál magasabbak.    
Tűzijáték minden hónap 3. napján a 4. emeleten.   

_*Minden szobához jár egy koktélkupon, amit a bárban lehet beváltani._   
_*A kupon értéke (BEG pénznemben) a házirendben szereplő számok szorzatának 2. 4. és 5. számjegyéből állítható össze. (A `.` tizedespontot jelöl)_


### <font color="lime">2. feladat - segíts Geekynek, és számítsd ki, hogy mennyi a koktélkupon legkedvezőbb értéke!</font>

In [2]:
szorzat = 13 * 11 * 6.12 * 21.41 * 1.37 * 5.02 * 3 * 4
indexek = [2, 4, 5]

# Szám stringé alakítása
szorzat_str = str(szorzat)

# Számjegyek kiválasztása
kivalasztott_szamjegyek = [szorzat_str[i-1] for i in indexek]

# Számjegyek rendezése csökkenő sorrendbe
kivalasztott_szamjegyek.sort(reverse=True)

# Számjegyekből szám létrehozása
kupon_max_erteke = int(''.join(kivalasztott_szamjegyek))

print(kupon_max_erteke)

653


### <font color="red">>>> 2. feladat megoldás - A koktélkupon legkedvezőbb értéke: 653 BEG</font>

Geeky a bárban a koktélját szürcsölte. Egy szemüveges bergengóc telepedett mellé.    
"A Koalai Harsona főszerkesztője vagyok. Mivel mi különösen becsüljük az éleseszű turistákat, szeretnék felajánlani egy lehetőséget.   
Ha itt helyben megoldja a következő feladványt, a Harsona befizeti egy kétnapos szafarira, ahol bergengóc tevéken ügethet, sünöket simogathat, oroszlánokkal kergetőzhet és fókákon lovagolhat."

Geeky szeme felcsillant. Nagyon szerette az állatokat, és a rejtvényekért is rajongott.   
Előhúzta Python-i-Max III. hordozható számítógépét, és feszülten figyelt.   

A főszerkesztő megköszörülte a torkát, előhúzott egy ív papírt a zsebéből, és hangosan olvasni kezdett:   
"Réges-régen három birodalom: a Ber, a Gen és a Góc királyság harcolt Bergengócia földjéért.      
A több generációnyi viszályt a három nagy király megelégelte, és békét kötött.    
Gündzselizeben, Krakatakiwimanga és Alfadelagunix megalapították Bergengóciát, ahol békében él egymás mellett ber, gen és góc.   
Ápolják különös hagyományaikat, melyek egyike az, hogy szigorú szabályoknak megfelelő neveket adnak az utódoknak.  

A berek neve 14 betűs. Ha előfordul benne a "dzse", akkor a név "en"-re vagy "em"-re végződik, egyébként "ka"-ra. A szóban az "i" legfeljebb egyszer fordulhat elő.   
A genek neve vagy 17 betűs, és "ga", "ge" vagy "gu" végződésű, vagy pedig 16 betűs, ekkor az utolsó két szótag mindenképpen "manga".     
A gócok neve mindig páratlan számú betűből áll. Ha 10 betűnél hosszabb, akkor az utolsó betű "x", egyébként "z".   
Bergengóciába sokan jöttek bevándorlók, de az ő neveik nem követik a fenti névadás hagyományait."

A főszerkesztő Geekyre nézett, szemében huncut fény csillant.   
"Nos, kisasszony, a kérdés, hogy szerkesztőségünk 15 tagja közül ki bergengócok, és ki bevándorlók leszármazottja?"   

Átnyújtott egy névlistát a lánynak, Geeky pedig gyorsan gépelni kezdett, s már futott is a Python program, amely megállapította, hogy a 15 névből melyek voltak hagyományos bergengócok. A főszerkesztő az eredmények láttán elégedetten bólintott, és elismerő szavakat dünnyögve egy elegáns, aranyszínű szafarijegyet adott a lánynak.

Szerkesztőségünk tagjai: <font color="magenta">Gedzsebrinxen, Kündzseperecem, Albinautixyx, Mereswedorian, Kilimoparatuka, Besteriendzsiken, Myrsortarian, Myrsatexeride, Kiribeterko, Fendzsupirjiken, Nowaroquartix, Miopelpetide, Werselenghider, Aldiretichewen, Broxadyrpex.</font>

### <font color="lime">3. feladat - kiket jelölt meg Geeky tősgyökeres bergengócként?</font>

In [3]:
szerkesztőség = ['Gedzsebrinxen', 'Kündzseperecem', 'Albinautixyx',
                 'Mereswedorian', 'Kilimoparatuka', 'Besteriendzsiken',
                 'Myrsortarian', 'Myrsatexeride', 'Kiribeterko',
                 'Fendzsupirjiken', 'Nowaroquartix', 'Miopelpetide',
                 'Werselenghider', 'Aldiretichewen', 'Broxadyrpex']

BERGEK = []
GENEK = []
GÓCOK = []

for név in szerkesztőség:
    
    #BERGEK kiválasztása a szerkesztőség tagjai közül
    if len(név) == 14 and név.count("i") <= 1:
        if "dzse" in név:
            if név.endswith("en") or név.endswith("em"):
                BERGEK.append(név)
        elif név.endswith("ka"):
            BERGEK.append(név)
    
    #GENEK kiválasztása a szerkesztőség tagjai közül
    if len(név) == 17:
        if név.endswith("ga") or név.endswith("ge") or név.endswith("gu"):
            GENEK.append(név)
    elif len(név) == 16:
        if név.endswith("manga"):
            GENEK.append(név)
            
    #GÓCOK kiválasztása a szerkesztőség tagjai közül
    if len(név) % 2 != 0:
        if len(név) > 10 and név.endswith("x"):
            GÓCOK.append(név)
        elif len(név) <= 10 and név.endswith("z"):
            GÓCOK.append(név)
    
print(BERGEK)
print(GENEK)
print(GÓCOK)

['Kündzseperecem']
[]
['Nowaroquartix', 'Broxadyrpex']


### <font color="red">>>> 3. feladat megoldás - a törzsgyökeres bergengócok a szerkesztőségben: 
### <font color="red">Kündzseperecem (BER származású), Nowaroquartix és Broxadyrpex (GÓC származásúak) </font>